### PR TITLE
[Shopify] - Different Shipping Charges Types Tests

### PR DESCRIPTION
**Pull request by GediminasGaubys**
This pull request does not have a related issue as it's part of the delivery for development agreed directly with @AndreiPanko

Different shipping charges types Tests
Automated tests for changes https://github.com/microsoft/ALAppExtensions/pull/26708

Fixes #26819
Fixes [AB#556525](https://dynamicssmb2.visualstudio.com/1fcb79e7-ab07-432a-a3c6-6cf5a88ba4a5/_workitems/edit/556525)

### CODE REVIEW

**Review by onbuyuka — APPROVED** [review]

File: Apps/W1/Shopify/test/Shipping/ShpfyShippingChargesTest.Codeunit.al [hunk 1]
```diff
@@ -0,0 +1,446 @@
+codeunit 139576 "Shpfy Shipping Charges Test"
```
Comment (JesperSchulz) [inline Apps/W1/Shopify/test/Shipping/ShpfyShippingChargesTest.Codeunit.al:1]: ID conflict. Please use 139546.

**Comment by JesperSchulz** [comment]
Internal run results in test failures:

Error: Cannot find an overload for "GetEnumerator" and the argument count: "0". Item:  
Error: Cannot find an overload for "GetEnumerator" and the argument count: "0". Item:  
Storing coverage results of  in:  S:\s\NAV\TestCoverageMap\TestCoverageMap_20241601_148191.dat
Success - Codeunit Integration Tests - Duration 34 seconds
   Success - Test method: SubmitDocument - Duration 8 seconds)
   Success - Test method: SubmitDocument_Pending_Sent - Duration 9 seconds)
   Success - Test method: SubmitDocument_Error_Sent - Duration 11 seconds)
   Success - Test method: SubmitDocumentAvalaraServiceDown - Duration 2 seconds)
   Success - Test method: OpenCompanyList - Duration 4 seconds)
Error: Cannot find an overload for "GetEnumerator" and the argument count: "0". Item:  
[2024-11-01 18:32:27] Test run completed
[2024-11-01 18:32:27] Failed tests:
[2024-11-01 18:32:27] Name: 
Method: UnitTestMapShippingChargesForEmptyType
Time: 0.204
Message: 
The VAT Posting Setup does not exist. Identification fields and values: VAT Bus. Posting Group='SSAZUWBNNH',VAT Prod. Posting Group='VAT7'
StackTrace: 
Sales Line(Table 37).VAT Prod. Posting Group - OnValidate(Trigger) line 15 - Base Application by Microsoft
Sales Line(Table 37).ValidateVATProdPostingGroup line 9 - Base Application by Microsoft
Sales Line(Table 37).No. - OnValidate(Trigger) line 115 - Base Application by Microsoft
Shpfy Process Order(CodeUnit 30166).CreateLinesFromShopifyOrder line 95 - Shopify Connector by Microsoft
Shpfy Process Order(CodeUnit 30166).OnRun(Trigger) line 15 - Shopify Connector by Microsoft
Shpfy Shipping Charges Test(CodeUnit 139546).UnitTestMapShippingChargesForEmptyType line 36 - Shopify Connector Test by Microsoft
Test Runner - Mgt(CodeUnit 130454).RunTests line 26 - Test Runner by Microsoft
Test Runner - Isol. Codeunit(CodeUnit 130450).OnRun(Trigger) line 4 - Test Runner by Microsoft
Test Suite Mgt.(CodeUnit 130456).RunTests line 2 - Test Runner by Microsoft
Test Suite Mgt.(CodeUnit 130456).RunSelectedTests line 35 - Test Runner by Microsoft
Test Suite Mgt.(CodeUnit 130456).RunNextTest line 20 - Test Runner by Microsoft
Command Line Test Tool(Page 130455).RunNextTest - OnAction(Trigger) line 7 - Test Runner by Microsoft


Name: 
Method: UnitTestMapShippingChargesForGLType
Time: 0.18
Message: 
The VAT Posting Setup does not exist. Identification fields and values: VAT Bus. Posting Group='SSAZUWBNNH',VAT Prod. Posting Group='VAT7'
StackTrace: 
Sales Line(Table 37).VAT Prod. Posting Group - OnValidate(Trigger) line 15 - Base Application by Microsoft
Sales Line(Table 37).ValidateVATProdPostingGroup line 9 - Base Application by Microsoft
Sales Line(Table 37).No. - OnValidate(Trigger) line 115 - Base Application by Microsoft
Shpfy Process Order(CodeUnit 30166).CreateLinesFromShopifyOrder line 92 - Shopify Connector by Microsoft
Shpfy Process Order(CodeUnit 30166).OnRun(Trigger) line 15 - Shopify Connector by Microsoft
Shpfy Shipping Charges Test(CodeUnit 139546).UnitTestMapShippingChargesForGLType line 38 - Shopify Connector Test by Microsoft
Test Runner - Mgt(CodeUnit 130454).RunTests line 26 - Test Runner by Microsoft
Test Runner - Isol. Codeunit(CodeUnit 130450).OnRun(Trigger) line 4 - Test Runner by Microsoft
Test Suite Mgt.(CodeUnit 130456).RunTests line 2 - Test Runner by Microsoft
Test Suite Mgt.(CodeUnit 130456).RunSelectedTests line 35 - Test Runner by Microsoft
Test Suite Mgt.(CodeUnit 130456).RunNextTest line 20 - Test Runner by Microsoft

**Comment by petemchlk** [comment]
pushed fix, can you re-run the internal tests? @JesperSchulz

**Review by JesperSchulz — DISMISSED** [review]

File: Apps/W1/Shopify/test/Shipping/ShpfyShippingChargesTest.Codeunit.al (hunk 1 above)
Comment (JesperSchulz) [inline Apps/W1/Shopify/test/Shipping/ShpfyShippingChargesTest.Codeunit.al:1]:
```suggestion
codeunit 139546 "Shpfy Shipping Charges Test"
```

**Comment by petemchlk** [comment]
@JesperSchulz missed one place where we used the same pattern for GL Account. I've pushed fix for this issue.

**Comment by JesperSchulz** [comment]
The test is still failing. Now with a slightly different message:

[2024-11-28 16:01:27] Failed tests:
[2024-11-28 16:01:27] Name: 
Method: UnitTestMapShippingChargesForGLType
Time: 0.216
Message: 
The VAT Posting Setup does not exist. Identification fields and values: VAT Bus. Posting Group='SSAZUWBNNH',VAT Prod. Posting Group='GU00000046'
StackTrace: 
Sales Line(Table 37).VAT Prod. Posting Group - OnValidate(Trigger) line 27 - Base Application by Microsoft
Sales Line(Table 37).ValidateVATProdPostingGroup line 9 - Base Application by Microsoft
Sales Line(Table 37).No. - OnValidate(Trigger) line 116 - Base Application by Microsoft
Shpfy Process Order(CodeUnit 30166).CreateLinesFromShopifyOrder line 92 - Shopify Connector by Microsoft
Shpfy Process Order(CodeUnit 30166).OnRun(Trigger) line 15 - Shopify Connector by Microsoft
Shpfy Shipping Charges Test(CodeUnit 139546).UnitTestMapShippingChargesForGLType line 38 - Shopify Connector Test by Microsoft
Test Runner - Mgt(CodeUnit 130454).RunTests line 26 - Test Runner by Microsoft
Test Runner - Isol. Codeunit(CodeUnit 130450).OnRun(Trigger) line 4 - Test Runner by Microsoft
Test Suite Mgt.(CodeUnit 130456).RunTests line 2 - Test Runner by Microsoft
Test Suite Mgt.(CodeUnit 130456).RunSelectedTests line 35 - Test Runner by Microsoft
Test Suite Mgt.(CodeUnit 130456).RunNextTest line 20 - Test Runner by Microsoft
Command Line Test Tool(Page 130455).RunNextTest - OnAction(Trigger) line 7 - Test Runner by Microsoft

**Comment by petemchlk** [comment]
@JesperSchulz pushed another fix, now it should be fine. Sorry for that it goes like this - it worked fine on our container all the time. It makes harder to find the issue.

**Comment by JesperSchulz** [comment]
> @JesperSchulz pushed another fix, now it should be fine. Sorry for that it goes like this - it worked fine on our container all the time. It makes harder to find the issue.

No worries! Let's try again :-)

**Review by JesperSchulz — APPROVED** [review]

**Comment by JesperSchulz** [comment]
Thanks for reporting this. We agree, and we’ll publish a fix asap, either in an update for the current version or in the next major release. Please do not reply to this, as we do not monitor closed issues. If you have follow-up questions or requests, please create a new issue where you reference this one.

Build ID: 27531.

**Pull request merged**